### ISSUE TITLE
torch 1.4.0 cannot find type2backend

### ISSUE THREAD

**Issue by pfeatherstone** [NVlabs/pacnet#14]
I think something as been shuffled around in torch. I cannot find type2backend which was previously in torch._thnn in 0.4.
Can we port this to torch 1.4.0?

**Comment by vkothapally** [comment]
I had the same issue. @suhangpro Do you have plans to make it compatible with PyTorch 1.4.0?

**Comment by suhangpro** [comment]
@pfeatherstone Sorry for the delay. Yes, we definitely plan to bring compatibility to 1.4. Will take a look soon (in the next few days or perhaps a week).

**Comment by vkothapally** [comment]
@suhangpro Just curious to know if you got a chance to look into this issue

**Comment by suhangpro** [comment]
There is now a branch "th14" which the test cases pass with pytorch 1.4. Not yet tested with any real experiments though.

**Comment by vkothapally** [comment]
Thanks a lot for the quick update @suhangpro

**Comment by jgsimard** [comment]
@suhangpro is there an update on this issue?

**Comment by Katou2** [comment]
it seems that ignoring "ctx._backend = type2backend[input.type()]" works fine, could we just ignore it in the new version of Pytorch. BTW, what are we doing by using "ctx._backend = type2backend[input.type()]"? Thanks. @suhangpro @vkothapally

**Comment by suhangpro** [comment]
@Katou2 Thanks for sharing the observation. That line was originally used in some custom layer examples for exposing low-level functions. If it runs without errors then I think you're right you've found an even easier fix!

**Comment by EyalRozenberg1** [comment]
Ignoring this import (as in https://github.com/NVlabs/pacnet/issues/14#issuecomment-622223277) seems to be working for the forward pass but raises errors in backprop.

Using the new branch (https://github.com/NVlabs/pacnet/issues/14#issuecomment-594341872) solved the issue.

**Comment by scarlett16** [comment]
> Ignoring this import (as in [#14 (comment)](https://github.com/NVlabs/pacnet/issues/14#issuecomment-622223277)) seems to be working for the forward pass but raises errors in backprop.
> 
> Using the new branch ([#14 (comment)](https://github.com/NVlabs/pacnet/issues/14#issuecomment-594341872)) solved the issue.

Sorry to bother you, but i don't know what the new branch(14th) is. 
Is that a new package or something?  Could you please post the link for me? Thank you so much!

**Comment by EyalRozenberg1** [comment]
> > Ignoring this import (as in [#14 (comment)](https://github.com/NVlabs/pacnet/issues/14#issuecomment-622223277)) seems to be working for the forward pass but raises errors in backprop.
> > Using the new branch ([#14 (comment)](https://github.com/NVlabs/pacnet/issues/14#issuecomment-594341872)) solved the issue.
> 
> Sorry to bother you, but i don't know what the new branch(14th) is.
> Is that a new package or something? Could you please post the link for me? Thank you so much!

Sure, no problem.
you can find branch th14 [here](https://github.com/NVlabs/pacnet/branches)

**Comment by EricPengShuai** [comment]
@suhangpro What is the main change of your new branch **th14** to solve the problem of type2backend’s missing, can you briefly talk about it? Thanks

**Comment by suhangpro** [comment]
@EricPengShuai You can check the changes [here](https://github.com/NVlabs/pacnet/commit/7ca0a3fc6b30bba195c1a288bfb80b90440e08e1). backend is no longer exposed and so we changed the implementation to just use the higher level `fold` function.
